### PR TITLE
parser: improve string parsing performance

### PR DESCRIPTION
**Pull request by belochub**
Improve string parsing performance by avoiding copying when possible.

### CODE REVIEW

File: src/jsrs_parser.cc [hunk 1]
```diff
@@ -456,8 +463,10 @@ MaybeLocal<Value> ParseString(Isolate*    isolate,
 
     if (begin[i] == '\\') {
       if (IsLineTerminatorSequence(begin + i + 1, &in_offset)) {
+        use_fallback(i - 1);
```
Comment (aqrln) [inline src/jsrs_parser.cc]: We can move this line to the outer block and get rid of the analogous one in the `else` clause, can't we?

Comment (belochub) [inline src/jsrs_parser.cc]: @aqrln
Nope, it won't work the same.

Comment (belochub) [inline src/jsrs_parser.cc]: @aqrln 
Yeah, after looking at it one more time it looks okay, I changed it like you proposed.

**Comment by aqrln** [comment]
Latest changes LGTM. Landed in 1f1d7d0a094a0785c8b5ebd144ca41e658e6cc7a.

**Pull request closed without merging**